### DISCUSSION TITLE
Chat

### DISCUSSION THREAD

**Discussion by Animeboynz** [Animeboynz/Mihon-Backup-Viewer#25] [General]
Just a chat

**Comment by Animeboynz** [comment]
I don't even know what to do anymore.
Gimme Ideas pleaseeeeeeeeeeeeee
@BrutuZ @NGB-Was-Taken

**Comment by BrutuZ** [comment]
Here's a thought: Seeing as the whole thing is client-side, when the mock-data is used the browser has to download it so I think it wouldn't be a bad idea to at least minify it and maybe replace it with a simpler one or even a tachibk version.

For the sake of argument, just minifying the current `data.json` shrinks it from 11.2MB to 7.88MB

**Reply by BrutuZ** [reply]
Worth noting that it would be especially egregious for me since I do all my testing in Private/Incognito/Ephemeral sessions.
Thankfully I do most of it locally, the few occasions I use the Netlify builds I'm mindful of avoiding the "Use Demo data" button and using local files instead

**Reply by Animeboynz** [reply]
That's Fair, the JSON data is just one of my old backups because I couldn't be bothered to create a dummy one at the time. And I chose the JSON instead of the .tachibk because the code has to convert it to JSON first before loading it which would take slightly longer. I will try find some time to craft some dummy data which is smaller and easier to load.

**Comment by Animeboynz** [comment]
Some sources don't have names? (from the python converter)
![image](https://github.com/Animeboynz/Mihon-Backup-Viewer/assets/40583749/7171190d-b688-43c6-998d-89d9a169bf91)

**Reply by Animeboynz** [reply]
Bruh, the website tachibk converter is rounding last 4 digits of the sourceId data.
![image](https://github.com/Animeboynz/Mihon-Backup-Viewer/assets/40583749/8db5e185-2ede-4aab-8ff7-0a33870b8846)

**Reply by BrutuZ** [reply]
> Some sources don't have names? (from the python converter) 

Can confirm, I also have one of those even extracting with ProtoC so it's not a decoder issue. My guess is uninstalled/obsolete extensions

EDIT: Seems about right, my nameless Source entry is for a source that was removed and then forbidden back in the days when they were built into Tachi, so wasn't ever installed even in the device I retired before the one that generated the backup.

The name may be fetched during the backup process directly from either the extension or the repo and probably doesn't recycle previously known ones.
Too bad the backup doesn't include the `lang` field too :P

**Reply by Animeboynz** [reply]
Yeah my one with a missing extension name is a source uninstalled a while back. So I think that if the extension isn't installed at backup time, it doesn't include its name. We have 2 options to handle this, 
1. Just show them as "Unknown Source"
2. Have a list of known `sourceId`s and resolve them on our end [Sources](https://gist.github.com/shashank-p/39b5d5ea9aeb523b7d3a97451d838f9d)

But I think the rounding of the `sourceId` is a way more of a serious issue.

**Comment by BrutuZ** [comment]
Unknown Source it is, don't think looking up elsewhere is a good idea.

The rounding has been solved already, I've referenced the commit in the issue

**Reply by Animeboynz** [reply]
Oh damb nice.

**Comment by NGB-Was-Taken** [comment]
@BrutuZ any idea why actions have been failing for the last two merges?

**Reply by BrutuZ** [reply]
Yep, I'd changed the action file on #38, fixing it now

**Comment by Animeboynz** [comment]
```
<3>WSL (9) ERROR: CreateProcessParseCommon:708: Failed to translate C:\Users\Animeboynz\Documents\GitHub\Mihon-Backup-Viewer
<3>WSL (9) ERROR: CreateProcessParseCommon:754: getpwuid(0) failed 2
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\app-3.4.1\resources\app\git\mingw64\libexec\git-core
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\app-3.4.1\resources\app\git\mingw64\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\app-3.4.1\resources\app\git\usr\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\app-3.4.1\resources\app\git\mingw64\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\app-3.4.1\resources\app\git\mingw64\usr\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Amazon Corretto\jdk21.0.2_13\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\ImageMagick-7.1.1-Q16-HDRI
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\system32
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\Wbem
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\WindowsPowerShell\v1.0
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\OpenSSH
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Microsoft VS Code\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\dotnet
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Docker\Docker\resources\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\GitHub CLI
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\nodejs
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Git\cmd
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\Microsoft\WindowsApps
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Local\GitHubDesktop\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\.dotnet\tools
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\IntelliJ IDEA 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\CLion 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\DataGrip 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\DataSpell 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\PhpStorm 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\PyCharm 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\RubyMine 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\JetBrains\WebStorm 2023.3.2\bin
<3>WSL (9) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\Animeboynz\AppData\Roaming\npm
<3>WSL (9) ERROR: CreateProcessEntryCommon:331: getpwuid(0) failed 2
<3>WSL (9) ERROR: CreateProcessEntryCommon:502: execvpe /bin/bash failed 2
<3>WSL (9) ERROR: CreateProcessEntryCommon:505: Create process not expected to return
husky - pre-commit script failed (code 1)
```
![image](https://github.com/Animeboynz/Mihon-Backup-Viewer/assets/40583749/4333da6d-f16e-454f-a22f-ef9abaed18a8)


I cant commit anymore 😭

**Reply by BrutuZ** [reply]
Was that after following https://github.com/Animeboynz/Mihon-Backup-Viewer/blob/main/CONTRIBUTING.md#formatting-your-code ?

I successfully tested a few dummy commits on my fork before pushing those

**Reply by Animeboynz** [reply]
@BrutuZ Yeah and I still cant commit

**Reply by BrutuZ** [reply]
Maybe some oddity of running trough WSL?

I've even cloned fresh on a separate drive and setup the pre-commit from scratch to try and make sure it worked before pushing it =/

**Reply by BrutuZ** [reply]
To be fair, it is working on the GH Action too 🤔

**Reply by BrutuZ** [reply]
In the worst case scenario that you can't debug it, there's also the option of disabling the pre-commit on your environment, there's still editor integration and GitHub Actions as fallbacks.
It's as simple as nuking the line `hooksPath = .husky/_` from your `.git/config` file

**Reply by Animeboynz** [reply]
Ok so here is what I have noticed. 
I deleted the repo from my PC and cloned it again and now I can commit.
But the second I run `npm ci` I can no longer commit. 
Also I don't use WSL so I'm not why that's part of the error unless the precommit script is using WSL.

My Fix for now.
My IDE has prettier built in so I'm just gonna manually format before I commit and I'm not gonna use npm to get any packages for the moment.

**Comment by Animeboynz** [comment]
Chores
-Refactor edit-menu to kebab-menu

**Comment by CypherVoid-NYX** [comment]
> Just a chat

how to use mihon in pc... im new plus... is there any app for it.. can you please tell me
